File: hackerrank/problem-solving/sales-by-match.js
function sockMerchant(n, ar) {
    const map = {};

    ar.forEach(val => {
        const key = val.toString();
        map[key] ? map[key] += 1 : map[key] = 1;
    });

    let count = 0;
    Object.values(map).forEach(val => {
        count += Math.floor(val / 2);
    });

    return count;
}

File: hackerrank/problem-solving/sales-by-match.cpp
int sockMerchant(int n, vector<int> ar) {
    map<int, int> m;
    int count = 0;

    for (int i = 0; i < n; i++) {
        m[ar[i]] += 1;
    }

    for (auto i = m.begin(); i != m.end(); i++) {
        count += i->second / 2;
    }

    return count;
}
